src/hooks/useShowCallback.js

```diff
@@ -7,18 +7,44 @@ import {
   isFunction
 } from 'min-dash';
 
-export function useShowCallback(businessObject, path) {
+/**
+ * Returns a memoized callback to be passed as `show` prop. Callback returns
+ * true if (1) ID and path match or (2) ID matches and matcher returns true.
+ *
+ * @example
+ *
+ * // using path
+ * const show = useShowCallback(businessObject, [ 'foo' ]);
+ *
+ * @example
+ *
+ * // using matcher
+ * const show = useShowCallback(businessObject, (event) => event.foo === 'bar');
+ *
+ * @param {Object} businessObject
+ * @param {string[]|Function} matcher
+ *
+ * @returns {Function}
```

Comment:
Should be Boolean isn't it?

Reply:
No, it returns a function. That function returns a boolean.

```js
const show = useShowCallback(); // function

show(); // boolean
```

Reply:
Ah sure!
sorry

Reply:
In this case the proper type signature would be `() => boolean`.